README.md

```diff
@@ -159,15 +159,26 @@ any errors encountered during the request.
 
 ## Body
 
-If the server's response contains a body it should follow the requirements for [GraphQL response](https://graphql.github.io/graphql-spec/June2018/#sec-Response).
+If the server's response contains a body it MUST follow the requirements for [GraphQL response](https://graphql.github.io/graphql-spec/June2018/#sec-Response).
 
-Note: For any non-2XX response, the client should not rely on the body to be in GraphQL format since the source of the response
-may not be the GraphQL server but instead some intermediary such as API gateways, firewalls, etc.
+Note: For any non-`200` response, the client should not rely on the body to be in GraphQL format.
+The source of the response may not be the GraphQL server, but instead some intermediary such as API gateways, firewalls, etc.
 
 ## Status Codes
 
-If the `data` entry in the response has any value other than `null` (when the operation has successfully executed
-without error) then the response should use the 200 (OK) status code.
-If the operation failed before or during execution, due to a syntax error, missing information, validation error
-or any other reason, then response should use the 4XX or 5XX status codes.
-It is recommended to use the same error codes as the [reference implementation](https://github.com/graphql/express-graphql).
+If the server successfully executed parts of the operation and returns a well-formed response
+that contains a non-null key `data`, the status code MUST be `200`.
+
+If the status code is `200`, the response MUST be a well-formed GraphQL response.
+
+The result of executing a GraphQL operation may contain partial data as well as encountered errors.
+Errors that happen during execution of the GraphQL operation become part of the result.
+
+In case of errors that completely prevent the successful execution of the request, the server SHOULD respond
+with the appropriate status code depending on the concrete error condition. For example:
+
+- `400` (Bad Request), in case of a validation error
+- `401` (Unauthorized), if the client is not allowed to access the schema or any of the requested fields
+- `500` (Internal Server Error), for unexpected failures in the server
+
```

Comment:
Do we want to also mention `403 (Forbidden)`?  This may be contentious in that authorization could be done at the field level.  I argue that some servers may also apply it at the schema/server level.

Reply:
Right, i don't think the examples need to be exhaustive. As you noted, this one is a bit tricky and might require a somewhat lengthy explanation in order to not be misleading.

On the other hand, authorization might be a great and illustrative example of an error condition that can be handled on the field level or on the HTTP level.

Reply:
https://github.com/APIs-guru/graphql-over-http/pull/79/commits/8a1108f0f5395e9f257dbaab050a0e4c18c08e94